src/Model/Search.php

```diff
@@ -29,4 +39,207 @@ public static function getUserTags()
 
 		return $tags;
 	}
+
+	/**
+	 * Search a user based on his/her profile address
+	 * pattern: @username@domain.tld
+	 *
+	 * @param string $user The user to search for
+	 *
+	 * @return ResultList|null
+	 * @throws \Friendica\Network\HTTPException\InternalServerErrorException
+	 * @throws \ImagickException
+	 */
+	public static function searchUser($user)
+	{
+		if ((filter_var($user, FILTER_VALIDATE_EMAIL) && Network::isEmailDomainValid($user)) ||
+		    (substr(Strings::normaliseLink($user), 0, 7) == "http://")) {
+
+			$user_data = Probe::uri($user);
+			if (empty($user_data)) {
+				return null;
+			}
+
+			if (!(in_array($user_data["network"], [Protocol::ACTIVITYPUB, Protocol::DFRN, Protocol::OSTATUS, Protocol::DIASPORA]))) {
+				return null;
+			}
+
+			$contactDetails = Contact::getDetailsByURL(defaults($user_data, 'url', ''), local_user());
+			$itemurl = (($contactDetails["addr"] != "") ? $contactDetails["addr"] : defaults($user_data, 'url', ''));
+
+			$result = new Result(
+				defaults($user_data, 'name', ''),
+				defaults($user_data, 'addr', ''),
+				$itemurl,
+				defaults($user_data, 'url', ''),
+				defaults($user_data, 'photo', ''),
+				defaults($user_data, 'network', ''),
+				defaults($contactDetails, 'cid', 0),
+				0,
+				defaults($user_data, 'tags', '')
+			);
+
+			return new ResultList(1, 1, 1, [$result]);
+
+		} else {
+			return null;
+		}
+	}
+
+	/**
+	 * Search in the global directory for occurrences of the search string
+	 * This is mainly based on the JSON results of https://dir.friendica.social
+	 *
+	 * @param string $search
+	 * @param int    $page
+	 *
+	 * @return ResultList|null
+	 * @throws \Friendica\Network\HTTPException\InternalServerErrorException
+	 */
+	public static function searchDirectory($search, $page = 1)
```

Comment:
Same gripe about a possible misplacement of this method. Maybe `Core\Search::getContactsFromGlobalDirectory`?